manifests/extract.pp

```diff
@@ -41,25 +41,30 @@
   $strip_components=0,
   $purge=false,
   $user=undef,
-  $tar_command=$::osfamily ? {
-    'Solaris' => 'gtar',
-    default   => 'tar',
-  },
+  $tar_command=undef,
 ) {
 
+  require ::archive::params
+
   if $root_dir {
     $extract_dir = "${target}/${root_dir}"
   } else {
     $extract_dir = "${target}/${name}"
   }
 
+  if $tar_command {
+    $real_tar_command = $tar_command
+  } else {
+    $real_tar_command = $::archive::params::tarcmd
+  }
+
```

Comment:
That's right, my bad. What you can do is make the define fail (using the `fail` function) if the class (or the parameter) is not defined.